File: client-js/src/controller/api/dto/AuthenticationDto.ts
/**
 * To authenticate a user, would need specific settings, like
 * specific values in the HTTP headers (eg, cookies)
 */
export default class AuthenticationDto {

    /**
     * The name given to this authentication info.
     * Just needed for display/debugging reasons
     */
    name: string;

    /**
     * The headers needed for authentication
     */
    headers = new Array<HeaderDto>();

    /**
     * If the login is based on cookies, need to provide info on
     * how to get such a cookie
     */
    cookieLogin: CookieLoginDto;


}

File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/auth/AuthenticationDto.java
package org.evomaster.client.java.controller.api.dto.auth;

import com.webfuzzing.commons.auth.AuthenticationInfo;

import java.util.ArrayList;
import java.util.List;

/**
 * To authenticate a user, would need specific settings, like
 * specific values in the HTTP headers (eg, cookies)
 */
public class AuthenticationDto extends AuthenticationInfo {

    /**
     * if the auth is processed based on RPC endpoints,
     * specify what info are required to execute the endpoint
     */
    public JsonAuthRPCEndpointDto jsonAuthEndpoint;

    /**
     * if the auth is processed with handleLocalAuthenticationSetup
     * specify what info is as input to setup auth with handleLocalAuthenticationSetup
     */
    public LocalAuthenticationDto localAuthSetup;


    public AuthenticationDto() {
    }

    public AuthenticationDto(String name) {
        this.setName(name);
    }
}
